solr/core/src/java/org/apache/solr/metrics/AggregateMetric.java

```diff
@@ -93,18 +115,11 @@ public double getMax() {
     if (values.isEmpty()) {
       return 0;
     }
-    Double res = null;
+    double res = 0;
```

Comment:
Is it guaranteed that values are going to be `>0`? how about a negatives?

Reply:
That's a good point - I don't know. (Who might know?)

It may be better to switch to creating a list of numerical values then finding its min/max/mean/etc (ideally using standard library functions) - but this may be overkill.

Reply:
Maybe @sigram ?

Reply:
I've added a (temporary) warning when negative numbers are encountered - this immediately gave me this in the log, which indicates my assumption was incorrect:

```
o.a.s.m.AggregateMetric negative metric value: -1.0
```

So I think I need to go back to the drawing board with this. Will review the tests and to ensure the negative number situation is covered.

Reply:
I have a breaking test now - have figured out how MetricUtilsTest works.